### DISCUSSION TITLE
Refactor the 'GuardClauseExtensions.cs'

### DISCUSSION THREAD

**Discussion by rafaelsc** [ardalis/GuardClauses#141] [Ideas]
The main extension file 'GuardClauseExtensions.cs' is big and long with many differently clauses mixed in the same file.

I would like to suggest dividing this file in many others smaller files. This will make new changes easer faster to implement and probably with less merge conflicts.

I can help with that. But first I would like to know what is @ardalis preferences.  
- Like one file per clause type, or (NullClauses.cs, OutOfRangeClauses.cs, etc)
- One file per target type. ( ArgumentClauses.cs, IncomparableClauses.cs, etc )
- or Mixed with both.

**Comment by ardalis** [comment]
Hi @rafaelsc 

I like the idea. I would use one file per clause type. So for instance all of the `Negative` methods would live in one file.

For naming what do you think of using something close to how they are used?

- `GuardAgainstNullExtensions.cs`
- `GuardAgainstNegativeExtensions.cs`
- etc.

?

**Comment by rafaelsc** [comment]
Hi @ardalis 

Like your idea too. I work on that when I have more free time. And fewer Open PRs to avoid merge conflicts.
